### ISSUE TITLE
[数据] 数据加载与 Dataset 封装

### ISSUE THREAD

**Issue by lingbai-i** [oom-alchemists/bert-Teams#12]
## 负责文件
`src/data.py`

## 任务
- 理解并负责数据加载模块：`load_txt` / `load_labels` / `IntentDataset` / `build_dataloader`
- 验证三种数据格式读取正确（train/dev/test，注意 CRLF 换行兼容）
- 答辩时需讲解 Dataset 三要素（`__init__`/`__len__`/`__getitem__`）与 DataLoader 的 batch_size/shuffle 作用

## 验收标准
- `python -c "from src.data import load_txt; print(len(load_txt('data/train.txt')[0]))` 输出 202500
- 能对 mini 训练的数据流脱稿讲解

## 时间
今晚完成代码理解与验证，明早 10:00 前如有改动走 PR 到 dev

**Comment by DaleFu22** [comment]
**data/class.txt ──load_labels──► 标签名列表（下标 = 标签ID）
data/*.txt     ──load_txt────► (texts, labels)  两个等长平行列表
                    │
                    └──subsample（mini 训练抽样）
                           │
                           └──build_dataloader──► IntentDataset ──► DataLoader**
### load_labels:把标签体系读成有序列表，行号即 ID，同时是分类头维度的唯一来源。
### load_txt:读训练/验证/测试数据
### IntentDataset:
      def __init__(self, texts, labels, tokenizer, max_len):   # 要素1：存数据与配置
      def __len__(self):            # 要素2：样本总数，决定一个 epoch 有几个 batch
      def __getitem__(self, i):     # 要素3：按下标产出单条样本
      两个必须讲清的技术点：
      squeeze(0) 为什么要：tokenizer 的 return_tensors="pt" 返回的是 [1, seq_len]（带 batch 维），但 __getitem__ 的契约是单条样本，所以 DataLoader 之后要能自动堆成 [B, seq_len]——必须先把那一维挤掉。
      padding="max_length" 的取舍：每条都 padding 到 48，好处是 batch 内形状绝对一致（torch.stack 不会失败）；代价是短句浪费。实测你们的文本 token 长度 max=20、均值 10.9，所以 48 里约 77% 是 padding。这是正确的工程取舍（换来了「不用写 collate_fn」的简洁），但可以和 DataCollatorWithPadding（动态 padding）对比着讲，是加分项。
      懒加载：__getitem__ 是取到哪条才 tokenize 哪条，不是一开始把 20 万条全编码进内存。这是 PyTorch Dataset 的核心契约，也是为什么它能吃下 20 万条数据。

### build_dataloader: 组装成可迭代的 batch
把 IntentDataset 包进 DataLoader，DataLoader 负责取样 → 组 batch → 自动 collate 成 [B, 48] 张量 → 迭代产出字典。
shuffle 的语义（唯一需要关注的参数）：
训练集传 True（train.py:86）——打乱避免模型学到样本顺序；
评估集传默认 False（train.py:87-88、distill.py:70、quantize.py:62）——保证评估结果可复现、可比。

### **总结：**
load_labels 从 class.txt 读出 9 个标签，行号就是标签 ID，同时决定分类头维度；load_txt 把「文本 + 标签 ID」读成两个平行列表，做了 CRLF 兼容、空行与脏标签处理，脏标签直接报错带行号而不是静默跳过；IntentDataset 实现 Dataset 三要素，按下标懒加载 tokenize，统一 padding 到 max_len=48 以保证 batch 形状一致；build_dataloader 把 Dataset 包成 DataLoader，训练打乱、评估不打乱。整条链路喂给 train / distill / prune / quantize 四个模块。

**Comment by lingbai-i** [comment]
已完成：`src/data.py` 随骨架 #21 入库，#29 又补充了空文本跳过与非法标签行号定位。验收命令输出 202500 已验证。
